functions/Test-DbaPowerPlan.ps1

```diff
@@ -51,7 +52,7 @@ function Test-DbaPowerPlan {
     #>
     param (
         [parameter(ValueFromPipeline)]
-        [DbaInstance[]]$ComputerName = $env:COMPUTERNAME,
```

Comment:
If this is not set, should it be mandatory as well?

Reply:
No, because I can pipe in from `Get-DbaPowerPlan` and in that case $computerName must not be set (so I had to remove the default).
If we remove InputObject here as well, then we would have to make $ComputerName mandetory. I don't see any added value in doing `Get-DbaPowerPlan -SqlInstance XYZ | Test-DabPowerPlan` because `Test-DabPowerPlan -SqlInstance XYZ` would be shorter. So if you agree, we can remove InputObject in all power plan commands and just accept ComputerName as piped input.

Reply:
Piping in the ComputerName from Get-DbaPowerPlan should replace the default so it is not required to remove it. 

We can remove InputObject if the functionality does not change. If I could go back in time, I'd probably just use the heck out of `ValueFromPipelineByPropertyName`.

Reply:
No problem with piping in ComputerName. But with piping in from Get- into InputObject sets the default to ComputerName and so not ony the computers from the pipeline are tested, but also the local computer. That's the reason for removing the default.

Reply:
Can you maybe add a comment down below with all the possible piping ways that should work? Then I have a todo list and can maybe create small mock commands for all three to show just the implementation of parameter processing.

Reply:
it would be 

```
        [parameter(ValueFromPipelineByPropertyName)]
        [DbaInstance[]]$ComputerName,
        [parameter(ValueFromPipelineByPropertyName)]
        [PSCredential]$Credential,
        [parameter(ValueFromPipelineByPropertyName)]
        [Alias("CustomPowerPlan")]
        [string]$PowerPlan,
        [switch]$EnableException
```

Then you can remove the InputObject!

Reply:
Not that easy:
![image](https://user-images.githubusercontent.com/66946165/124244108-e1e03080-db1e-11eb-8df6-919d3d234917.png)

Reply:
This works: 
```
[parameter(ValueFromPipelineByPropertyName, ValueFromPipeline)]
[DbaInstance[]]$ComputerName,
```

Reply:
Implemented that, see comment down below.